events/abc/client.go

```diff
@@ -0,0 +1,137 @@
+package abc
+
+import (
+	"github.com/checkout/checkout-sdk-go/client"
+	"github.com/checkout/checkout-sdk-go/common"
+	"github.com/checkout/checkout-sdk-go/configuration"
+)
+
+type Client struct {
+	configuration *configuration.Configuration
+	apiClient     client.HttpClient
+}
+
+func NewClient(configuration *configuration.Configuration, apiClient client.HttpClient) *Client {
+	return &Client{
+		configuration: configuration,
+		apiClient:     apiClient,
+	}
+}
+
+func (c *Client) RetrieveAllEventTypes(
+	query QueryRetrieveAllEventType,
```

Comment:
Seems that this name is odd and probably is not needed, since we cannot default to `nil` parameters in GO, I'd say to create two functions for this particular scenario, because in your tests you are not considering that version may be nil, then the response should return events for version 1.0 & 2.0

Reply:
I get it. I will modify it